src/ape_cache/query.py

```diff
@@ -0,0 +1,336 @@
+                    return 200
+                # Can't handle this query
+                # TODO: Allow partial queries
+                return None
+
+        except Exception as err:
+            # Note: If any error, skip the data from the cache and continue to
+            #       query from provider.
+            logger.debug(err)
+            return None
+
+    @singledispatchmethod
+    def perform_query(self, query: QueryType) -> pd.DataFrame:  # type: ignore
+        raise QueryEngineError(f"Cannot handle '{type(query)}'.")
+
+    @perform_query.register
+    def perform_block_query(self, query: BlockQuery) -> pd.DataFrame:
+        with self.engine.connect() as conn:
+            q = conn.execute(
+                text(
+                    """
+                    SELECT :columns
+                    FROM blocks
+                    WHERE blocks.number >= :start_block
+                    AND blocks.number <= :stop_block
+                    AND blocks.number mod :step = 0
+                    """
+                ),
+                columns=",".join(query.columns),
+                start_block=query.start_block,
+                stop_block=query.stop_block,
+                step=query.step,
+            )
+            return pd.DataFrame(columns=query.columns, data=q.fetchall())
+
+    @perform_query.register
+    def perform_transaction_query(self, query: BlockTransactionQuery) -> pd.DataFrame:
+        with self.engine.connect() as conn:
+            q = conn.execute(
+                text(
+                    """
+                    SELECT :columns
+                    FROM transactions
+                    WHERE transactions.block_id = :block_id
+                    """
+                ),
+                columns=",".join(query.columns),
+                start_block=query.block_id,
+            )
+            return pd.DataFrame(columns=query.columns, data=q.fetchall())
+
+    @perform_query.register
+    def perform_contract_events_query(self, query: ContractEventQuery) -> pd.DataFrame:
+        with self.engine.connect() as conn:
+            q = conn.execute(
+                text(
+                    """
+                    SELECT :columns
+                    FROM contract_events
+                    WHERE contract_events.block_number >= :start_block
+                    AND contract_events.block_number <= :stop_block
+                    AND contract_events.block_number mod :step = 0
+                    """
+                ),
+                columns=",".join(query.columns),
+                start_block=query.start_block,
+                stop_block=query.stop_block,
+                step=query.step,
+            )
+            return pd.DataFrame(columns=query.columns, data=q.fetchall())
+
+    @singledispatchmethod
+    def update_cache(self, query: QueryType) -> None:  # type: ignore
+        raise QueryEngineError(f"Cannot handle '{type(query)}'.")
```

Comment:
Do nothing in the common case, because basically this doesn't support caching unsupported queries, and this method shouldn't through an exception at all
```suggestion
        pass
```